contracts/vesting/src/storage.rs

```diff
@@ -0,0 +1,189 @@
+};
+
+use crate::error::ContractError;
+
+impl TryFromVal<Env, DataKey> for Val {
+    type Error = ConversionError;
+
+    fn try_from_val(_env: &Env, v: &DataKey) -> Result<Self, Self::Error> {
+        Ok((*v as u32).into())
+    }
+}
+
+#[derive(Clone, Copy)]
+#[repr(u32)]
+pub enum DataKey {
+    Admin = 1,
+    Config = 2,
+    Minter = 3,
+    Whitelist = 4,
+    VestingTokenInfo = 5,
+    MaxVestingComplexity = 6,
+}
+
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct VestingTokenInfo {
+    pub name: String,
+    pub symbol: String,
+    pub decimals: u32,
+    pub address: Address,
+    pub total_supply: i128,
+}
+
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct VestingBalance {
+    pub address: Address,
+    pub balance: i128,
+    pub curve: Curve,
+}
+
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct MinterInfo {
+    pub address: Address,
+    pub capacity: Curve,
+}
+
+#[contracttype]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct VestingInfo {
+    pub amount: i128,
+    pub curve: Curve,
+}
+
+pub fn save_admin(env: &Env, admin: &Address) {
+    env.storage().persistent().set(&DataKey::Admin, admin);
+}
+
+pub fn get_admin(env: &Env) -> Address {
+    env.storage()
+        .persistent()
+        .get(&DataKey::Admin)
+        .unwrap_or_else(|| {
+            log!(&env, "Vesting: Get admin: Critical error - No admin found");
+            panic_with_error!(env, ContractError::NoAdminFound);
+        })
+}
+
+pub fn save_balance(env: &Env, address: &Address, balance: i128) {
+    env.storage().persistent().set(address, &balance);
+}
+
+pub fn save_vesting(env: &Env, address: &Address, vesting_info: &VestingInfo) {
+    env.storage().instance().set(address, vesting_info);
+}
+
+pub fn get_vesting(env: &Env, address: &Address) -> Result<VestingInfo, ContractError> {
+    let vesting_info = env.storage().instance().get(address).unwrap_or_else(|| {
+        log!(&env, "Vesting: Get vesting schedule: Critical error - No vesting schedule found for the given address");
+        panic_with_error!(env, ContractError::VestingNotFoundForAddress);
+    });
+
+    Ok(vesting_info)
+}
+
+pub fn remove_vesting(env: &Env, address: &Address) {
+    env.storage().instance().remove(&address);
+}
+
+// TODO: uncomment when needed
+// pub fn get_allowances(env: &Env, owner_spender: &(Address, Address)) -> i128 {
+//     env.storage().persistent().get(owner_spender).unwrap_or_else(|| {
+//             log!(&env, "Vesting: Get allowance: Critical error - No allowance found for the given address pair");
+//             panic_with_error!(env, ContractError::AllowanceNotFoundForGivenPair);
+//         })
+// }
+
+// pub fn save_allowances(env: &Env, owner_spender: &(Address, Address), amount: i128) {
+//     env.storage().persistent().set(owner_spender, &amount);
+// }
+
+pub fn save_minter(env: &Env, minter: MinterInfo) {
+    env.storage().persistent().set(&DataKey::Minter, &minter);
+}
```

Comment:
Modify this in all access helpers. It not efficient to require a copy in an argument only to pass a reference later.

```suggestion
pub fn save_minter(env: &Env, minter: &MinterInfo) {
    env.storage().persistent().set(&DataKey::Minter, minter);
}
```